### PR TITLE
Report dag

### PR DESCRIPTION
**Pull request by whitleykeith**
Creates report dag to generate enriched summary docs from test runs

### CODE REVIEW

File: dags/reports/pipeline/metrics/control_plane.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+from reports.pipeline.metrics import util
+
+
+def get_average_cpu_of_control_plane_apps(report, prom_client):
+    result_set = {}
+
+    start = util.parse_timestamp(report['metadata']['start_date'])
+    end = util.parse_timestamp(report['metadata']['end_date'])
+    cluster = report['cluster_name']
+    labels = f'name!="", openshift_cluster_name="{cluster}", container!="POD",namespace=~"openshift-(etcd|oauth-apiserver|.*apiserver|ovn-kubernetes|sdn|ingress|authentication|.*controller-manager|.*scheduler|monitoring|image-registry|operator-lifecycle-manager)"'
```
[the remark this answers is not in the document]
Comment (whitleykeith) [reply dags/reports/pipeline/metrics/control_plane.py]: i think eventually we may want to add that to the config section. For now I changed the namespace selector to grab all namespaces prefixed with openshift

File: dags/reports/pipeline/metrics/control_plane.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+from reports.pipeline.metrics import util
+
+
+def get_average_cpu_of_control_plane_apps(report, prom_client):
+    result_set = {}
+
+    start = util.parse_timestamp(report['metadata']['start_date'])
+    end = util.parse_timestamp(report['metadata']['end_date'])
+    cluster = report['cluster_name']
+    labels = f'name!="", openshift_cluster_name="{cluster}", container!="POD",namespace=~"openshift-.*"'
```
[the remark this answers is not in the document]
Comment (whitleykeith) [reply dags/reports/pipeline/metrics/control_plane.py:12]: for sure, but i've decided to leave that for a future pr to keep this one small

File: dags/reports/pipeline/metrics/control_plane.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        }
+    else:
+        return {}
+
+def get_average_available_memory_of_master_nodes(report, prom_client):
+    start = util.parse_timestamp(report['metadata']['start_date'])
+    end = util.parse_timestamp(report['metadata']['end_date'])
+    cluster = report['cluster_name']
+    master_nodes = get_masters(report, prom_client)
+    labels = f'openshift_cluster_name="{cluster}", instance=~"{"|".join(master_nodes)}", mode!="idle"'
```
Comment (jtaleric) [inline dags/reports/pipeline/metrics/control_plane.py:72]: Do we need the mode for memory? Should it be Available?

File: dags/openshift_nightlies/scripts/run_benchmark.sh [hunk 4]
```diff
@@ -88,7 +88,7 @@ else
         export OPENSHIFT_MUST_GATHER=true
         export STORAGE_MODE=snappy
         export WORKLOAD=$AIRFLOW_CTX_TASK_ID-must-gather
-        ./ocp_diagnosis.sh || true
+        ./ocp_diagnosis.sh > /dev/null 2>&1
```
[the remark this answers is not in the document]
Comment (whitleykeith) [reply dags/openshift_nightlies/scripts/run_benchmark.sh]: implemented both of the requested changes. Only redirect stdout and added a conditional to only run must-gather on a failure of the benchmark

File: dags/reports/tasks/generate.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    grafana_url = Variable.get('grafana')
+    thanos_url = Variable.get('thanos_querier_url')
+
+    git_user = util.get_git_user()
+
+    if git_user == "cloud-bulldozer":
+        target_index = "ci-reports"
+    else: 
+        target_index = f"{git_user}-reports"
+        config['searchIndices'].append(f"{git_user}_playground")
```
[the remark this answers is not in the document]
Comment (whitleykeith) [reply dags/reports/tasks/generate.py:44]: no it'll just not have metric data in the report (just an empty field)

**Pull request merged**